samples/bluetooth/bap_broadcast_sink/src/audio_codec.c

```diff
@@ -0,0 +1,172 @@
+/**
+ * @file
+ * @brief Bluetooth BAP Broadcast Sink audio codec integration
+ *
+ * This file handles all the audio codec hardware integration for the sample
+ *
+ * Copyright (c) 2025 SiFli Technologies(Nanjing) Co., Ltd
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <zephyr/autoconf.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/audio/codec.h>
+
+LOG_MODULE_REGISTER(codec, CONFIG_LOG_DEFAULT_LEVEL);
+
+/*
+ * Size of one audio block transferred to the codec, in bytes.
+ *
+ * For the configuration used here (16bit samples, mono channel),
+ * 480 bytes correspond to 240 PCM samples per block. This value was
+ * chosen as a compromise between:
+ *  - keeping latency reasonably low (smaller blocks)
+ *  - avoiding excessive interrupt / DMA overhead (larger blocks).
+ * If the PCM format (sample width or number of channels) is changed,
+ * this constant should be revisited so that the block size remains
+ * appropriate for the target use case and hardware.
+ */
+#define CODEC_BLOCK_SIZE 480
+
+/*
+ * Default speaker/output volume sent via AUDIO_PROPERTY_OUTPUT_VOLUME.
+ *
+ * The valid numeric range and exact mapping to dB are codec‑dependent;
+ * refer to the specific codec driver / datasheet for details. The value
+ * 15 was selected as a conservative, comfortable default for this sample
+ * application. Adjust as needed to match the desired default output level.
+ */
+#define SPEAKER_VOL      15
```

Comment:
Consider making this a Kconfig option if you want people to adjust it. Ideally it would server as the default volume value and then users could modify it via buttons on a device or even using the Volume Control Profile, although I think dynamically adjusting the value is outside the scope of this PR